LibertyDSNP/example-client
# Feature/embedded url/177294663

Purpose
---------------
Feature

[https://www.pivotaltracker.com/story/show/177294663](https://www.pivotaltracker.com/story/show/177294663)

As a user, when I include a link in the text of a post or reply, it should be clickable.  OnClick, the link is opened in a new tab.


Solution
---------------
When reading the content for posts and replies we want to look for valid URL, replace them with a link tag.

styling:
if a valid URL, it should turn the URL to dark green.


Change summary
---------------
* add a find valid link helper function
* use when displaying content for a post or reply
* add testing


Steps to Verify
----------------
1. Put a link in a caption
2. put a link in a reply
3. click on it
4. see that it takes you to a new page


Additional details / screenshot
----------------
- it was too much of a pain to auto find the URL when you're typing and change the styling, so we just kept that black and only inserted the colored link in the saved post and replies

<img width="781" alt="Screen Shot 2021-11-02 at 12 23 47 PM" src="https://user-images.githubusercontent.com/43625033/139933132-8388997b-3176-4412-8b88-2277ba363021.png">
